EventFilter/CTPPSRawToDigi/plugins/TotemVFATRawToDigi.cc

```diff
@@ -168,9 +169,19 @@ template <typename DigiType>
 void TotemVFATRawToDigi::run(edm::Event &event, const edm::EventSetup &es) {
   // get DAQ mapping
   ESHandle<TotemDAQMapping> mapping = es.getHandle(totemMappingToken);
+  if (!mapping) {
+    LogError("TotemVFATRawToDigi") << "TotemVFATRawToDigi: no mapping found for " << subSystemName;
+  }
 
   // get analysis mask to mask channels
-  ESHandle<TotemAnalysisMask> analysisMask = es.getHandle(analysisMaskToken);
+  TotemAnalysisMask analysisMask;
+  ESHandle<TotemAnalysisMask> analysisMaskHandle = es.getHandle(analysisMaskToken);
+  if (analysisMaskHandle) {
+    analysisMask = *analysisMaskHandle;
+  } else {
+    LogWarning("TotemVFATRawToDigi") << "TotemVFATRawToDigi: no mask found for " << subSystemName;
+    analysisMask = TotemAnalysisMask();
```

Comment:
what are the consequences of using a default constructed object in this case?

Reply:
I am still curious to know what happens in case the conditions are missing and the empty object is used.

Reply:
actually, so far it has always been an empty object, because we've never (yet) used a mask

Reply:
I am not sure if that is a right place to ask, but maybe we can get some advice here.
Is there any good practice or policy for handling log messages in case products are missing in the CondDB ? That could happen due to misconfiguration (of GT ?) or simply by human mistake. 

For some of the products, like `TotemAnalysisMask`  we can safely handle the situation by creating an empty object. As @brzozia has mentioned this object is set to empty for all the T0 reconstruction modules for PPS . It is only used for private local reconstruction. 

We've added here a log message. If for some reason the `TotemAnalysisMask` won't be found in CondDB then we will see a log message for every event. 

I've seen that in July 2023 the PR 42303 introduced a smart way of handling log messages for CTPPSPixelRawToDigi using a new class CTPPSPixelErrorSummary. After the changes some logs are not printed per event but after closing the stream. This way we avoid the pollution of the log files.

Related code is here:
https://github.com/cms-sw/cmssw/pull/42303/files#diff-6159c1ddb281e81b4e708480289d84db4b7a87347d58c77af36e12324666ea7d

The change was requested and discussed in this issue:
https://github.com/cms-sw/cmssw/issues/41456#issuecomment-1632197861

I was considering reusing the CTPPSPixelErrorSummary approach here (for example by renaming CTPPSPixelErrorSummary -> CTPPSRawToDigiErrorSummary ) and using it for collection of the messages.

Reply:
To be honest, no one really looks at tier0 logs in a systematic way. In my opinion the only good way of dealing with missing products is:
   * if the product is fundamental (i.e. without it the job will lead to nonsensical results) fail the job (i.e. throw an exception)
   * if the job still makes sense, don't pollute the logs with a printout for each event.  Let me remind that we're not doing debugging in production and that huge logs cost Tier0 disk.

Reply:
OK, that makes sense, then we proceed with smarter log handling. Do you consider this PR a good place for such improvement or should go in a separate PR ?

Reply:
> OK, that makes sense, then we proceed with smarter log handling. Do you consider this PR a good place for such improvement or should go in a separate PR ?

Please proceed here for whatever is related to the code updated by this PR. Other files can be touched in a separate PR